### PR TITLE
Fix passing a `Request` instance as input

### LINKED ISSUE #134: Passing a Request object as first argument to ky fails
I don't know if this is intentional, but the README - and the TypeScript definitions - state that in `ky(input, options)` `ky` accepts the same values as `fetch`.

However, passing a Request instance as the value for `input` does not work, since it's stringified [here](https://github.com/sindresorhus/ky/blob/master/index.js#L171) and it turns into `[object Request]`.

### PR DESCRIPTION
**Pull request by whitecrownclown**
Fixes #134 

Followed up on https://github.com/sindresorhus/ky/issues/134#issuecomment-491044682 and also added a test for it.

### CODE REVIEW

**Comment by sindresorhus** [comment]
This doesn't correctly solve it. `Request` is not just the URL, otherwise it would have been a point-less class, as we already have URL. `Request` can contain headers, the body, etc.

https://developer.mozilla.org/en-US/docs/Web/API/Request

**Comment by sholladay** [comment]
Indeed, what I meant to say in the issue is that we'll need to inspect the request object's URL when constructing the `prefuxUrl`. But the request object has other data, such as the headers, that we also need to respect.

**Comment by whitecrownclown** [comment]
> This doesn't correctly solve it. `Request` is not just the URL, otherwise it would have been a point-less class, as we already have URL. `Request` can contain headers, the body, etc.
> 
> [developer.mozilla.org/en-US/docs/Web/API/Request](https://developer.mozilla.org/en-US/docs/Web/API/Request)

@sindresorhus 
How do we want it to work? Deconstruct the `Request` instance into `ky` options or just pass it on as `input`?

To be more specific:

```js
const requestOptions = {headers: {Referer: 'http://google.com'}, method: 'POST'};
const myRequest = new Request('http://example.com', requestOptions);

(async () => {
	await ky(myRequest, {method: 'GET'}); // which method is used?
})();
```

**Comment by sindresorhus** [comment]
> await ky(myRequest, {method: 'GET'}); // which method is used?

`myRequest` should take priority. Options should modify it. We should clearly document this fact though.

I assume that's how `fetch` works too. Can you check how `fetch` behaves in such situation? Doesn't it silently ignore the options, or does it throw an error?

**Comment by whitecrownclown** [comment]
`fetch` does take the options into account. 

> const requestOptions = {headers: {Referer: 'http://google.com'}, method: 'POST'};
const myRequest = new Request('http://example.com', requestOptions);

> (async () => {
	await ky(myRequest, {method: 'GET'}); // which method is used?
})();

`GET` is used here.

**Comment by sindresorhus** [comment]
Hmm, interesting. Then I think we should follow what `fetch` does. We've tried to do that with other features too.

File: index.js [hunk 1]
```diff
@@ -195,6 +195,8 @@ class Ky {
 			this._input = url.toString();
 		}
 
+		this._input = input instanceof Request ? input : this._input;
```
Comment (szmarczak) [inline index.js]: What about moving this to above? @sindresorhus 

```js
if (input instanceof Request) {
    this._input = input;
} else {
    this._input = String(input || '');
    ....
}

**Comment by whitecrownclown** [comment]
@sindresorhus 	This also means that any `url` related options from `ky` don't work when passing in a `Request` since we can't alter the `Request` instance url.

**Comment by sindresorhus** [comment]
Yes, just document that limitation.

File: index.js [hunk 2]
```diff
@@ -157,6 +157,45 @@ class Ky {
 			...otherOptions
 		};
 
+		if (input instanceof Request) {
```
Comment (sindresorhus) [inline index.js]: You need to do `getGlobal` for `Request`.

**Pull request merged**